src/sentry/api/endpoints/organization_shortids.py

```diff
@@ -0,0 +1,74 @@
+from __future__ import absolute_import
+
+from rest_framework.response import Response
+
+from django.db import transaction
+
+from sentry.api.bases.organization import OrganizationEndpoint
+from sentry.models import Project
+from sentry.utils.strings import validate_callsign
+
+
+class ShortIdsUpdateEndpoint(OrganizationEndpoint):
+
+    def put(self, request, organization):
+        """
+        Update Short IDs
+        ````````````````
+
+        Updates the call signs of projects within the organization.
+
+        :pparam string organization_slug: the slug of the organization the
+                                          short ID should be looked up in.
+        :param callsigns: a dictionary of project IDs to their intended
+                          callsigns.
+        :auth: required
+        """
+        callsigns = request.DATA.get('callsigns', {})
+        for project_id, callsign in callsigns.iteritems():
+            callsign = validate_callsign(callsign)
+            if callsign is None:
+                return Response({'detail': 'invalid callsign "%s"' % callsign},
+                                status=400)
+            callsigns[project_id] = callsign
+
+        if len(callsigns) != len(callsigns.values()):
+            return Response({'detail': 'Duplicate callsigns'}, status=400)
+
+        project_q = organization.project_set.filter(
+            pk__in=[int(x) for x in callsigns if x.isdigit()]
+        )
```

Comment:
- Why would `x` ever *not* be a digit here if it's represents the project ID?
- Wouldn't it be an error if the result set size wasn't the same as the input set size, since something intended to be updated isn't going actually going to be affected?